tlslite/messages.py

```diff
@@ -1765,3 +1759,64 @@ def parse(self, p):
 
     def write(self):
         return self.bytes
+
+
+class Heartbeat(object):
+    """
+    Handling Heartbeat extension from RFC 6520
+
+    @type message_type: int
+    @ivar message_type: type of message (response or request)
+
+    @type payload: bytearray
+    @ivar payload: payload
+
+    @type padding: bytearray
+    @ivar padding: random padding of selected length
+    """
+
+    def __init__(self):
+        self.contentType = ContentType.heartbeat
+        self.message_type = 0
+        self.payload = bytearray(0)
+        self.padding = bytearray(0)
+
+    def create(self, message_type, payload, padding_length):
+        """Create heartbeat request or response with selected parameters"""
+        self.message_type = message_type
+        self.payload = payload
+        self.padding = getRandomBytes(padding_length)
+        return self
+
+    def createResponse(self):
+        """Creates heartbeat response based on request."""
+        heartbeat_response = Heartbeat().create(
+            HeartbeatMessageType.heartbeat_response,
+            self.payload,
+            16)
+        return heartbeat_response
+
+    def parse(self, p):
+        """Deserialize heartbeat message from parser."""
+        self.message_type = p.get(1)
+        self.payload = p.getVarBytes(2)
+        return self
```

Comment:
saying in a comment that we're explicitly ignoring leftover bytes (padding) would make it a bit easier to know what's happening